<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>443</width>
    <height>397</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Energy</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="edge_group_box">
     <property name="toolTip">
      <string>Change the edge of the beamline by sending a plan to the queue server. This mechanism is meant for large moves to different X-ray absorption edges.

This will move the X-ray source (if applicable), change the harmonic, and move any monochromators. It will also attempt to maintain alignment to the extent possible.</string>
     </property>
     <property name="title">
      <string>Edge</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <layout class="QGridLayout" name="edge_layout">
        <item row="2" column="0">
         <widget class="QLabel" name="label_20">
          <property name="text">
           <string>Harmonic:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Energy:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="edge_combo_box">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>120</width>
            <height>0</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Sans Serif</family>
           </font>
          </property>
          <property name="toolTip">
           <string>Select an edge, or type one in the form &quot;Na L3&quot;, etc.</string>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="editable">
           <bool>true</bool>
          </property>
          <property name="currentText">
           <string>Ni K (8333 eV)</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_12">
          <item>
           <widget class="QueueserverButton" name="set_energy_button">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="toolTip">
             <string>Queue a plan to change the X-ray energy.</string>
            </property>
            <property name="text">
             <string>Set Energy</string>
            </property>
            <property name="willRunPlan" stdset="0">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item row="3" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_15">
          <item>
           <widget class="QCheckBox" name="offset_checkbox">
            <property name="text">
             <string/>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="offset_widget" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QDoubleSpinBox" name="offset_spinbox">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="offset_auto_checkbox">
               <property name="text">
                <string>Auto</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_21">
          <property name="text">
           <string>ID Offset:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="target_energy_spinbox">
          <property name="suffix">
           <string> eV</string>
          </property>
          <property name="maximum">
           <double>9999999999999.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QCheckBox" name="harmonic_checkbox">
            <property name="text">
             <string/>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="harmonic_widget" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="spacing">
              <number>5</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QSpinBox" name="harmonic_spinbox">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="singleStep">
                <number>2</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="harmonic_auto_checkbox">
               <property name="text">
                <string>Auto</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="energy_group_box">
     <property name="toolTip">
      <string>Changes the energy of the beamline over a narrow range. Best when staying within a single X-ray absorption edge.

It will change the monochromator and undulator (if applicable), but makes no attempt to keep the beam aligned and will not change the harmonic.</string>
     </property>
     <property name="title">
      <string>Energy</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <layout class="QFormLayout" name="energy_layout">
        <property name="labelAlignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>Set:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QDoubleSpinBox" name="move_energy_devices_spinbox">
            <property name="suffix">
             <string> ev</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="move_energy_devices_button">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Set</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_14">
          <item>
           <widget class="QPushButton" name="jog_reverse_button">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="jog_value_spinbox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string>How much to tweak the signal with each tweak.</string>
            </property>
            <property name="suffix">
             <string> ev</string>
            </property>
            <property name="maximum">
             <double>100000000000000000620008645040778319495168.000000000000000</double>
            </property>
            <property name="stepType">
             <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="jog_forward_button">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Monochromator</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_16">
          <item>
           <widget class="PyDMLabel" name="PyDMLabel_10">
            <property name="toolTip">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="PyDMRelatedDisplayButton" name="PyDMRelatedDisplayButton_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="PyDMIcon" stdset="0">
             <string>wrench</string>
            </property>
            <property name="filenames" stdset="0">
             <stringlist>
              <string>motor.py</string>
             </stringlist>
            </property>
            <property name="titles" stdset="0">
             <stringlist>
              <string>More</string>
             </stringlist>
            </property>
            <property name="showIcon" stdset="0">
             <bool>true</bool>
            </property>
            <property name="macros" stdset="0">
             <stringlist>
              <string>MOTOR=energy.monochromator.pitch2</string>
             </stringlist>
            </property>
            <property name="openInNewWindow" stdset="0">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_22">
          <property name="text">
           <string>Undulator</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_17">
          <item>
           <widget class="PyDMLabel" name="PyDMLabel_11">
            <property name="toolTip">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="PyDMRelatedDisplayButton" name="PyDMRelatedDisplayButton_4">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="PyDMIcon" stdset="0">
             <string>wrench</string>
            </property>
            <property name="filenames" stdset="0">
             <stringlist>
              <string>motor.py</string>
             </stringlist>
            </property>
            <property name="titles" stdset="0">
             <stringlist>
              <string>More</string>
             </stringlist>
            </property>
            <property name="showIcon" stdset="0">
             <bool>true</bool>
            </property>
            <property name="macros" stdset="0">
             <stringlist>
              <string>MOTOR=energy.monochromator.pitch2</string>
             </stringlist>
            </property>
            <property name="openInNewWindow" stdset="0">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PyDMLabel</class>
   <extends>QLabel</extends>
   <header>pydm.widgets.label</header>
  </customwidget>
  <customwidget>
   <class>PyDMRelatedDisplayButton</class>
   <extends>QPushButton</extends>
   <header>pydm.widgets.related_display_button</header>
  </customwidget>
  <customwidget>
   <class>QueueserverButton</class>
   <extends>QPushButton</extends>
   <header>firefly.queue_button</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources/beamline_components.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>harmonic_auto_checkbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>harmonic_spinbox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>104</y>
    </hint>
    <hint type="destinationlabel">
     <x>339</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>offset_auto_checkbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>offset_spinbox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>205</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>offset_checkbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>offset_widget</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>89</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>165</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>harmonic_checkbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>harmonic_widget</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>89</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>99</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
